storageClient/storageClient.go

```diff
@@ -113,6 +127,60 @@ func (c *StorageClient) UpsertNoteByIDString(ctx context.Context, params Note) (
 	return result, err
 }
 
+// CreateGroupMembershipKey creates an encrypted version of the provided groupkey for the specified group member.
+func (c *StorageClient) CreateGroupMembershipKey(ctx context.Context, params CreateMembershipKeyRequest) (string, error) {
+	var wrappedEncryptedMembershipKey string
+	// Decrypt the access key
+	rawEncryptionKey, err := e3dbClients.DecodeSymmetricKey(c.EncryptionKeys.Private.Material)
+	if err != nil {
+		return wrappedEncryptedMembershipKey, err
+	}
+	groupKey, err := e3dbClients.DecryptEAK(params.EncryptedGroupKey, params.ShareePublicKey, rawEncryptionKey)
+	if err != nil {
+		return wrappedEncryptedMembershipKey, err
+	}
+	// Using the decrypted version of
+	// encrypted Group Key and the public key of
+	// the group admin specified in params,
+	// create an encrypted membership key that
+	// the new member can decrypt
+	wrappedEncryptedMembershipKey, err = c.encryptMembershipKeyForGroupMember(ctx, groupKey, params.NewMemberID)
+	return wrappedEncryptedMembershipKey, err
+}
+
+// WrapMembershipKeyForGroupMember tales the group key and wraps it for the new client
+func (c *StorageClient) encryptMembershipKeyForGroupMember(ctx context.Context, groupKey e3dbClients.SymmetricKey, newMemberID string) (string, error) {
```

Comment:
comment needs to be updated for the new method name and please make this a public method `Encrypt....` so it can be used by other libraries.